src/box/memtx_space.c

```diff
@@ -1171,14 +1196,18 @@ memtx_build_on_replace_rollback(struct trigger *trigger, void *event)
 	return 0;
 }
 
-/*
- * Struct to allocate by memtx_build_on_replace trigger
- * (on_rollback trigger with its data).
+/**
+ * Deletes the txn triggers from memtx_ddl_state::txn_triggers
+ * since they are going to be deleted.
  */
-struct on_rollback_trigger_with_data {
-	struct trigger on_rollback;
-	struct index_build_on_rollback_data data;
-};
+static int
+memtx_build_on_replace_commit(struct trigger *trigger, void *event)
+{
+	(void)event;
+	struct memtx_build_txn_triggers *data = trigger->data;
+	rlist_del_entry(data, in_state);
+	return 0;
+}
```

Comment:
We usually use `container_of` instead of passing a pointer to self in the trigger argument. The code below would look more familiar and clean to me:

```C
static int
memtx_build_on_replace_commit(struct trigger *base, void *event)
{
	(void)event;
	struct memtx_build_stmt_trigger *trigger =
		container_of(base, struct memtx_build_stmt_trigger, on_commit);
	rlist_del_entry(trigger, in_state);
	return 0;
}
```

Reply:
Hm, OK, done - now I don't pass any data to the trigger.